lib/detail/view/qr_detail_screen.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mobile_test/app/dependencies.dart';
+import 'package:mobile_test/base/result_state.dart';
+import 'package:mobile_test/detail/cubit/qr_detail_cubit.dart';
+import 'package:mobile_test/widgets/loading_indicator.dart';
+import 'package:mobile_test/widgets/timer_text.dart';
+import 'package:qr_api/qr_api.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qr_models/qr_models.dart';
+
+class QrDetailScreen extends StatefulWidget {
+  const QrDetailScreen({Key? key}) : super(key: key);
+
+  static Widget init() {
+    final repository = getIt.get<SeedRepository>();
+    return BlocProvider(
+      create: (_) => QrDetailCubit(repository),
+      child: const QrDetailScreen(),
+    );
+  }
+
+  @override
+  State<QrDetailScreen> createState() => _QrDetailScreenState();
+}
+
+class _QrDetailScreenState extends State<QrDetailScreen> {
+  @override
+  void initState() {
+    super.initState();
+    context.read<QrDetailCubit>().getQrSeed();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('QR Code'),
+      ),
+      body: BlocConsumer<QrDetailCubit, ResultState<QRSeed>>(
+        builder: (context, state) {
+          return state.when(
+            initial: () => Container(),
+            loading: () => const LoadingIndicator(),
+            data: (data) => QrDetailSuccessPage(seed: data),
+            error: (e) => ErrorPage(message: e),
+          );
+        },
+        listener: (context, state) {
+          if (state is Error) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              const SnackBar(
+                content: Text('Something goes wrong :('),
+              ),
+            );
+          }
+        },
+      ),
+    );
+  }
+}
+
+class ErrorPage extends StatelessWidget {
```

Comment:
suggestion:  This error page worked well but isn't very helpful and would be confusing to a user.  Letting the user know there is a network error over a `dio error` might help them understand that they don't have an internet connection.  

![Screen Shot 2021-10-12 at 4 23 18 PM](https://user-images.githubusercontent.com/85515499/137023926-02515d37-b668-4068-a024-ae6de03231fb.png)